index/field/docs_field_test.go

```
@@ -259,6 +260,87 @@ func TestDocsFieldMergeInPlace(t *testing.T) {
+		intArrayPool    = pool.NewBucketizedIntArrayPool(intArrayBuckets, nil)
+		int64ArrayPool  = pool.NewBucketizedInt64ArrayPool(int64ArrayBuckets, nil)
+		stringArrayPool = pool.NewBucketizedStringArrayPool(stringArrayBuckets, nil)
+		boolAllocs      int
+		doubleAllocs    int
+		intAllocs       int
+		int64Allocs     int
+		stringAllocs    int
+	)
+	boolArrayPool.Init(func(capacity int) []bool {
+		boolAllocs++
+		return make([]bool, 0, capacity)
+	})
+	doubleArrayPool.Init(func(capacity int) []float64 {
+		doubleAllocs++
+		return make([]float64, 0, capacity)
+	})
+	intArrayPool.Init(func(capacity int) []int {
+		intAllocs++
+		return make([]int, 0, capacity)
+	})
+	int64ArrayPool.Init(func(capacity int) []int64 {
+		int64Allocs++
+		return make([]int64, 0, capacity)
+	})
+	stringArrayPool.Init(func(capacity int) []string {
+		stringAllocs++
+		return make([]string, 0, capacity)
+	})
+	opts := NewDocsFieldBuilderOptions().
+		SetBoolArrayPool(boolArrayPool).
+		SetDoubleArrayPool(doubleArrayPool).
+		SetIntArrayPool(intArrayPool).
+		SetInt64ArrayPool(int64ArrayPool).
+		SetStringArrayPool(stringArrayPool)
+	builder := NewDocsFieldBuilder([]string{"testPath"}, opts)
+
+	builder.Add(1, field.ValueUnion{Type: field.BoolType, BoolVal: true})
+	builder.Add(2, field.ValueUnion{Type: field.BoolType, BoolVal: false})
+	builder.Add(1, field.ValueUnion{Type: field.DoubleType, DoubleVal: 1})
+	builder.Add(2, field.ValueUnion{Type: field.DoubleType, DoubleVal: 2})
+	builder.Add(1, field.ValueUnion{Type: field.IntType, IntVal: 1})
+	builder.Add(2, field.ValueUnion{Type: field.IntType, IntVal: 2})
+	builder.Add(1, field.ValueUnion{Type: field.TimeType, TimeNanosVal: 1})
+	builder.Add(2, field.ValueUnion{Type: field.TimeType, TimeNanosVal: 2})
+	builder.Add(1, field.ValueUnion{Type: field.StringType, StringVal: "foo"})
+	builder.Add(2, field.ValueUnion{Type: field.StringType, StringVal: "bar"})
+	builder.Close()
+
+	assert.Equal(t, 1, boolAllocs)
```

Mind replacing `assert` with `require`? I'm a big fan of failing early.